extension/js/common/browser/browser-msg.ts

```diff
@@ -242,7 +255,7 @@ export class BrowserMsg {
     notificationShow: (dest: Bm.Dest, bm: Bm.NotificationShow) => BrowserMsg.sendCatch(dest, 'notification_show', bm),
     notificationShowAuthPopupNeeded: (dest: Bm.Dest, bm: Bm.NotificationShowAuthPopupNeeded) =>
       BrowserMsg.sendCatch(dest, 'notification_show_auth_popup_needed', bm),
-    showConfirmation: (dest: Bm.Dest, bm: Bm.ShowConfirmation) => BrowserMsg.sendCatch(dest, 'confirmation_show', bm),
+    showConfirmation: (bm: Bm.ShowConfirmation) => BrowserMsg.sendToParentWindow('confirmation_show', bm, 'confirmation_result'),
```

Comment:
Let's rename this to `sendToParentWindowUnsafely` ?

Is this truly unsafe? What is the issue - does host page js receive this too?

Reply:
yes, a handler from the host js or another extension installed to this page will be able to see the contents of this message

Reply:
I wonder if we could just encrypt & sign these messages, and then use this as the messaging without worry? Both receiving frame and sending frame will have access to local storage, where they can find the same symmetric password, let's say.

Then we can brutally simplify all of that browser messaging dispatch spaghetti.